SimPEG/VRM/ProblemVRM.py

```diff
@@ -155,170 +139,387 @@ def _getGeometryMatrix(self, xyzc, xyzh, pp):
-                        np.log(np.sqrt(u2**2+v1**2+w1**2)-u2) +
-                        np.log(np.sqrt(u2**2+v2**2+w1**2)-u2) -
-                        np.log(np.sqrt(u1**2+v2**2+w1**2)-u1) +
-                        np.log(np.sqrt(u1**2+v2**2+w2**2)-u1) -
-                        np.log(np.sqrt(u1**2+v1**2+w2**2)-u1) +
-                        np.log(np.sqrt(u2**2+v1**2+w2**2)-u2) -
-                        np.log(np.sqrt(u2**2+v2**2+w2**2)-u2)
-                    )
-
-                    Gzz = (
-                        - np.arctan((v1*w1)/(u1*np.sqrt(u1**2+v1**2+w1**2)+tol)) +
-                        np.arctan((v1*w1)/(u2*np.sqrt(u2**2+v1**2+w1**2)+tol)) -
-                        np.arctan((v2*w1)/(u2*np.sqrt(u2**2+v2**2+w1**2)+tol)) +
-                        np.arctan((v2*w1)/(u1*np.sqrt(u1**2+v2**2+w1**2)+tol)) -
-                        np.arctan((v2*w2)/(u1*np.sqrt(u1**2+v2**2+w2**2)+tol)) +
-                        np.arctan((v1*w2)/(u1*np.sqrt(u1**2+v1**2+w2**2)+tol)) -
-                        np.arctan((v1*w2)/(u2*np.sqrt(u2**2+v1**2+w2**2)+tol)) +
-                        np.arctan((v2*w2)/(u2*np.sqrt(u2**2+v2**2+w2**2)+tol))
-                    )
-
-                    Gzz = (
-                        Gzz -
-                        np.arctan((u1*w1)/(v1*np.sqrt(u1**2+v1**2+w1**2)+tol)) +
-                        np.arctan((u2*w1)/(v1*np.sqrt(u2**2+v1**2+w1**2)+tol)) -
-                        np.arctan((u2*w1)/(v2*np.sqrt(u2**2+v2**2+w1**2)+tol)) +
-                        np.arctan((u1*w1)/(v2*np.sqrt(u1**2+v2**2+w1**2)+tol)) -
-                        np.arctan((u1*w2)/(v2*np.sqrt(u1**2+v2**2+w2**2)+tol)) +
-                        np.arctan((u1*w2)/(v1*np.sqrt(u1**2+v1**2+w2**2)+tol)) -
-                        np.arctan((u2*w2)/(v1*np.sqrt(u2**2+v1**2+w2**2)+tol)) +
-                        np.arctan((u2*w2)/(v2*np.sqrt(u2**2+v2**2+w2**2)+tol))
-                    )
-
-                    G[COUNT, :] = C*np.c_[Gxz, Gyz, Gzz]
-                    COUNT = COUNT + 1
+            if isinstance(rxObj, Point):
+
+                if dComp.lower() == 'x':
+                    for rr in range(0, M):
+                        u1 = locs[rr, 0] - ax
+                        u1[np.abs(u1) < tol] = np.min(xyzh[:, 0])/tol2
+                        u2 = locs[rr, 0] - bx
+                        u2[np.abs(u2) < tol] = -np.min(xyzh[:, 0])/tol2
+                        v1 = locs[rr, 1] - ay
+                        v1[np.abs(v1) < tol] = np.min(xyzh[:, 1])/tol2
+                        v2 = locs[rr, 1] - by
+                        v2[np.abs(v2) < tol] = -np.min(xyzh[:, 1])/tol2
+                        w1 = locs[rr, 2] - az
+                        w1[np.abs(w1) < tol] = np.min(xyzh[:, 2])/tol2
+                        w2 = locs[rr, 2] - bz
+                        w2[np.abs(w2) < tol] = -np.min(xyzh[:, 2])/tol2
+
+                        Gxx = (
+                            np.arctan((v1*w1)/(u1*np.sqrt(u1**2+v1**2+w1**2)+tol)) -
```

Comment:
lots of those lines are too long for the liking of my linter. Do you use one?

Reply:
Looked through these. Made some spacing adjustments